src/tools/generators.ts

```diff
@@ -245,3 +259,279 @@ export function installGenerators(generators: string[]): string[] {
+      if (isMissing) return false
+
+      const metadata = await sharp(filePath).metadata()
+      return metadata.width !== 1024 || metadata.height !== 1024
+    })()
+
+    return { fileName, isMissing, isInvalidSize }
+  })
+
+  const validationResults = await Promise.all(validationPromises)
+
+  // accumulate error messages for any failed validations
+  const validationMessages = validationResults.reduce((acc, r) => {
+    const messages = [
+      r.isMissing && "   • missing",
+      r.isInvalidSize && "   • wrong size (expected 1024x1024px)",
+    ].filter(Boolean)
+
+    if (messages.length) {
+      acc.push(`⚠️  ${r.fileName}:`, ...messages)
+    }
+
+    return acc
+  }, [])
+
+  return {
+    isValid: !validationMessages.length,
+    messages: validationMessages,
+  }
+}
+
+/**
+ * Generates app-icons for specified option.
+ */
+export async function generateAppIcons(option: `${Platforms}` | "all") {
+  const { path, exists, find, copy, write } = filesystem
+  const cwd = process.cwd()
+
+  const options = option === "all" ? Object.values(Platforms) : ([option] as `${Platforms}`[])
+
+  const optionGenerationSuccesses = []
+
+  // start the generation process for each platform
+  // looping instead of mapping allows us to await for each platform sequentially
+  for (const o of options) {
+    const optionProjectName = { ios: "iOS", android: "Android", expo: "Expo" }[o]
+
+    // find the output path for platform and check if it exists
+    // iOS is a bit weird since it's named differently for each project
+    const relativeOutputDirPath = {
+      expo: "assets/images",
+      android: "android/app/src/main/res",
+      ios:
+        (exists("ios") &&
+          find("ios", {
+            directories: true,
+            files: false,
+            matching: "AppIcon.appiconset",
+          })?.[0]) ||
+        "ios/**/Images.xcassets/AppIcon.appiconset",
+    }[o]
+    const outputDirPath = path(cwd, relativeOutputDirPath)
+
+    // if not, skip...
+    if (exists(outputDirPath) !== "dir") {
+      warning(
+        `⚠️  No output directory found for "${optionProjectName}" at "${outputDirPath}". Skipping...`,
+      )
+      continue
+    }
+
+    heading(`Generating ${optionProjectName} app icons...`)
+
+    const icons = APP_ICON_RULES.icons.filter((i) => i.platform === o)
+
+    // prepare each icon for generation sequentially
+    for (const i of icons) {
+      const inputFilePath = path(templatesDir(), "app-icon", i.inputFile)
+
+      // get the input file for sharp and do some initial transforms when necessary (e.g. radius and padding)
+      const inputFile = await (async function () {
+        if (!i.transform) return inputFilePath
+
+        try {
+          const { size, radius, padding } = i.transform
+          const cutoutMask = Buffer.from(
+            `<svg><rect x="0" y="0" width="${size}" height="${size}" rx="${radius}" ry="${radius}"/></svg>`,
+          )
+          return await sharp(inputFilePath)
```

Comment:
```suggestion
          return sharp(inputFilePath)
```

minor nit -- when you return a promise, it's [recommended](https://dmitripavlutin.com/return-await-promise-javascript/) to skip the await and just `return sharp(inputFilePath)` here.

Reply:
You're completely right that I did it for the try..catch. Returning `await`s just looks weird IMO. Perhaps I can make this more readable by first assigning to a variable and returning that.